products/accessapproval/api.yaml

```diff
@@ -0,0 +1,240 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: AccessApproval
+display_name: Access Approval
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://accessapproval.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Access Approval
+    url: https://console.cloud.google.com/apis/library/accessapproval.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: FolderSettings
+    base_url: "folders/{{folder_id}}/accessApprovalSettings"
+    self_link: 'folders/{{folder_id}}/accessApprovalSettings'
+    create_verb: :PATCH
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Access Approval enables you to require your explicit approval whenever Google support and engineering need to access your customer content.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      api: 'https://cloud.google.com/access-approval/docs/reference/rest/v1/folders'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: folder_id
+        description: |
+          ID of the folder of the access approval settings.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        description: |
+          The resource name of the settings. Format is "folders/{folder_id}/accessApprovalSettings"
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: notificationEmails
+        description: |
+          A list of email addresses to which notifications relating to approval requests should be sent.
+          Notifications relating to a resource will be sent to all emails in the settings of ancestor
+          resources of that resource. A maximum of 50 email addresses are allowed.
+        max_size: 50
+        item_type: Api::Type::String
+      - !ruby/object:Api::Type::Array
+        name: enrolledServices
```

Comment:
I'm a bit nervous having this as a list instead of a set, since order shouldn't make a difference here and lists of objects get weird when adding/removing entries to/from the middle. Since it doesn't have any O+C fields, having it as a set should be safe, I think.